src/bundles/communication/RpcController.ts

```diff
@@ -0,0 +1,123 @@
+import { type MultiUserController } from './MultiUserController';
+import uniqid from 'uniqid';
+
+type DeclaredFunction = {
+  name: string;
+  func: (...args: any[]) => any;
+};
+
+/**
+ * Controller for RPC communication between 2 devices.
+ *
+ * @param topicHeader Topic header for all RPC communication, must not include '/'.
+ * @param multiUser Instance of multi user controller.
+ * @param userId ID of the user, used for identifying caller/callee.
+ */
+export class RpcController {
+  private topicHeader: string;
+  private multiUser: MultiUserController;
+  private userId: string;
+  private functions = new Map<string, DeclaredFunction>();
+  private pendingReturns = new Map<string, (args: any[]) => void>();
+  private returnTopic: string;
+
+  constructor(
+    topicHeader: string,
```

Comment:
Are we going to allow arbitrary, user-defined MQTT topics? This might pose a problem with our own AWS MQTT broker, specifically, due to the whitelist. Though this shouldn't be an issue for other brokers, but just something to take note.

Reply:
Yes, we need to advise students accordingly. That's beyond this PR.